bertopic/backend/_spacy.py

```diff
@@ -77,18 +77,21 @@ def embed(self,
         # empty strings to the zero vector
         empty_document = " "
 
-        # Extract embeddings from a transformer model
-        if "transformer" in self.embedding_model.component_names:
-            embeddings = []
-            for doc in tqdm(documents, position=0, leave=True, disable=not verbose):
-                embeddings.append(self.embedding_model(doc or empty_document)._.trf_data.tensors[-1][0].tolist())
-            embeddings = np.array(embeddings)
-
-        # Extract embeddings from a general spacy model
-        else:
-            embeddings = []
-            for doc in tqdm(documents, position=0, leave=True, disable=not verbose):
-                embeddings.append(self.embedding_model(doc or empty_document).vector)
-            embeddings = np.array(embeddings)
-
-        return embeddings
\ No newline at end of file
+        # Extract embeddings
+        embeddings = []
+        for doc in tqdm(documents, position=0, leave=True, disable=not verbose):
+            embedding = self.embedding_model(doc or empty_document)
+            if embedding.has_vector:
+                embedding = embedding.vector
+                it not isinstance(embedding, np.ndarray) and hasattr(embedding, 'get'):
+                    # Convert cupy array to numpy array
+                    embedding = embedding.get()
```

Comment:
Seeing as this and the one below are identical, perhaps putting them outside of the if/else statement might be more appropriate. Also, it should be `if` instead of `it`.

Reply:
Oops, silly mistake. Thanks! I updated this.

Reply:
Speaking of, maybe we can get a linter + CI testing set up in this repo. If you create an issue for it, I could try to help, if you want :)

Reply:
Oh, I realized they already are setup, I just didn't see them failing the PR like had expected. Nvm